python/lsst/pipe/analysis/colorAnalysis.py

```
@@ -363,281 +564,1057 @@ def plotGalaxyColors(self, catalogs, filenamer, dataId):
+        nKeepOdr = np.sum(keepOdr)
+        if nKeepOdr < order:
+            raise RuntimeError(
+                "Not enough good data points ({0:d}) for polynomial fit of order {1:d}".
+                format(nKeepOdr, order))
+        orthRegCoeffs = orthogonalRegression(xx[keepOdr], yy[keepOdr], order, initialGuess)
+    yOrthLine = np.polyval(orthRegCoeffs, xLine)
+    axes[0].plot(xLine, yOrthLine, "g--")
+
     kwargs = dict(s=3, marker="o", lw=0, alpha=0.4)
-    axes[0].scatter(xx[keep], yy[keep], c="blue", label="used", **kwargs)
-    axes[0].scatter(xx[~keep], yy[~keep], c="black", label="other", **kwargs)
+    axes[0].scatter(xx[~keep], yy[~keep], c=zOther, cmap="gray", label="other", **kwargs)
+    axes[0].scatter(xx[keep], yy[keep], c=zKeep, cmap="jet", label="used", **kwargs)
     axes[0].set_xlabel(xLabel)
     axes[0].set_ylabel(yLabel, labelpad=-1)
-    axes[0].legend(loc="upper left", fontsize=7)
-    axes[0].plot(xLine, yLine, "r-")
+
+    mappableKeep = plt.cm.ScalarMappable(cmap="jet", norm=plt.Normalize(vmin=zKeep.min(), vmax=zKeep.max()))
+    mappableKeep._A = []        # fake up the array of the scalar mappable. Urgh...
+    caxKeep = plt.axes([0.46, 0.15, 0.022, 0.75])
+    cbKeep = plt.colorbar(mappableKeep, cax=caxKeep)
+    cbKeep.ax.tick_params(labelsize=6)
+    labelPadShift = len(str(zKeep.max()//10)) if zKeep.max()//10 > 0 else 0
+    cbKeep.set_label("Number Density", rotation=270, labelpad=-15 - labelPadShift, fontsize=7)
+
+    # Find index where poly and fit range intersect -- to calculate the local slopes of the fit to make
+    # sure it is close to the fitLines (log a warning if they are not within 5%)
+    message = ("{0:s} branch of the hard-coded lines for object selection does not cross the "
+               "current polynomial fit.\nUsing the xFitRange {1:} to compute the local slope")
+    try:
+        crossIdxUpper = (np.argwhere(np.diff(np.sign(yOrthLine - yLineUpper)) != 0).reshape(-1) + 0)[0]
+    except:
```

bare except. Try
```
except Exception as e:
```
Bonus if you make Exception the actual Exception that you're expecting here.  Also, loggers are special and will supply the arguments to the string for you like `log.warnf(message, "Upper", xFitRange[1])` <<-- you can even use new style formatting with `warnf`: https://developer.lsst.io/stack/logging.html

---

I'm not sure of all the kinds of exceptions that could occur here, but I don't want to bail if the "find point of intersection" attempt fails, so I just added the **Exception**.